libs/utils/include/utils/compiler.h

```diff
@@ -64,6 +64,12 @@
 #    define UTILS_PRIVATE
 #endif
 
+#define UTILS_HAS_CXX17 0
+#if __cplusplus >= 201703L  || (defined(_MSVC_LANG) && _MSVC_LANG >= 201703L)
+#undef UTILS_HAS_CXX17
+#define UTILS_HAS_CXX17 1
+#endif
```

Comment:
confused about this one because we require C++17. So this should always be true.

Reply:
I think the goal is for clients to be able to compile against Filament public headers with older versions of MSVC.

Reply:
I wanted to ask/check the required C++ language version in Filament; I am glad to hear it is C++17.

And @bejado is right about my original intention. However I checked the public headers of Filament and they contain C++17 features (nested namespace definition, `std::string_view`) so it is not a valid use case to compile against Filament public headers with `/std:c++14` (or lower). So I removed UTILS_HAS_CXX17  in f8c74de3507e1861f7433a392b25fad69c4cbc04.

Reply:
Fun fact: _MaterialBuilder.h_ and _DriverEnums.h_ use designated initializers too ([which is a C++ 20 feature](https://en.cppreference.com/w/cpp/language/aggregate_initialization#Designated_initializers)).